core/src/test/java/com/ericsson/bss/cassandra/ecchronos/core/repair/state/TestAlarmPostUpdateHook.java

```diff
@@ -0,0 +1,271 @@
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestAlarmPostUpdateHook
+{
+    private static final String keyspaceName = "keyspace";
+    private static final String tableName = "table";
+
+    private static final long RUN_INTERVAL_IN_DAYS = 1;
+    private static final long GC_GRACE_DAYS = 10;
+
+    @Mock
+    private RepairStateSnapshot myRepairStateSnapshot;
+
+    @Mock
+    private RepairFaultReporter myFaultReporter;
+
+    private MockedClock myClock = new MockedClock();
+    private AlarmPostUpdateHook myPostUpdateHook;
+
+    private final TableReference myTableReference = new TableReference(keyspaceName, tableName);
+
+    @Before
+    public void startup()
+    {
+        RepairConfiguration repairConfiguration = RepairConfiguration.newBuilder()
+                .withRepairWarningTime(RUN_INTERVAL_IN_DAYS * 2, TimeUnit.DAYS)
+                .withRepairErrorTime(GC_GRACE_DAYS, TimeUnit.DAYS)
+                .build();
+        myPostUpdateHook = new AlarmPostUpdateHook(myTableReference, repairConfiguration, myFaultReporter);
+
+        myPostUpdateHook.setClock(myClock);
+    }
+
+    @Test
+    public void testThatWarningAlarmIsSentAndCeased()
+    {
+        // setup - not repaired
+        long daysSinceLastRepair = 2;
+        long start = System.currentTimeMillis();
+        long lastRepaired = start - TimeUnit.DAYS.toMillis(daysSinceLastRepair);
+
+        Map<String, Object> expectedData = new HashMap<>();
+        expectedData.put(RepairFaultReporter.FAULT_KEYSPACE, keyspaceName);
+        expectedData.put(RepairFaultReporter.FAULT_TABLE, tableName);
+
+        // mock - not repaired
+        doReturn(lastRepaired).when(myRepairStateSnapshot).lastRepairedAt();
+        myClock.setTime(start);
+
+        myPostUpdateHook.postUpdate(myRepairStateSnapshot);
+
+        // verify - not repaired
+        verify(myFaultReporter).raise(eq(RepairFaultReporter.FaultCode.REPAIR_WARNING), eq(expectedData));
+
+        // setup - repaired
+        lastRepaired = start;
+        start = System.currentTimeMillis();
+
+        // mock - repaired
+        doReturn(lastRepaired).when(myRepairStateSnapshot).lastRepairedAt();
+        myClock.setTime(start);
+
+        myPostUpdateHook.postUpdate(myRepairStateSnapshot);
+
+        // verify alarm ceased in preValidate
+        verify(myFaultReporter).cease(eq(RepairFaultReporter.FaultCode.REPAIR_WARNING), eq(expectedData));
+        reset(myFaultReporter);
+
+        myPostUpdateHook.postUpdate(myRepairStateSnapshot);
+
+        // verify - repaired
+        verify(myFaultReporter).cease(eq(RepairFaultReporter.FaultCode.REPAIR_WARNING), eq(expectedData));
+    }
+
+    @Test
+    public void testThatWarningAlarmIsSentAndCeasedExternalRepair()
```

Comment:
This is basically a duplicate of the previous test case (but shorter), we should remove it.